crates/prover/src/examples/poseidon/mod.rs

```diff
@@ -410,9 +413,43 @@ mod tests {
     };
     use crate::math::matrix::{RowMajorMatrix, SquareMatrix};
 
+    #[cfg(all(target_family = "wasm", not(target_os = "wasi")))]
+    wasm_bindgen_test::wasm_bindgen_test_configure!(run_in_browser);
+
+    #[cfg(feature = "parallel")]
+    #[cfg(all(target_family = "wasm", not(target_os = "wasi")))]
+    fn get_hardware_concurrency_js_sys() -> usize {
+        use js_sys::wasm_bindgen::{JsCast, JsValue};
+
+        let global = js_sys::global().unchecked_into::<js_sys::Object>();
+        let nav = js_sys::Reflect::get(&global, &JsValue::from_str("navigator"))
+            .ok()
+            .and_then(|v| v.dyn_into::<js_sys::Object>().ok())
+            .unwrap_or_default();
+        let hc = js_sys::Reflect::get(&nav, &JsValue::from_str("hardwareConcurrency"))
+            .ok()
+            .and_then(|v| v.as_f64())
+            .unwrap_or(1.0);
+        hc as usize
+    }
+
+    #[cfg(feature = "parallel")]
```

Comment:
We might want to always have an "init", even if trivial.

That way the calling code does not change except changing the feature flag, i.e. the interface is the same.